spark_log_parser/parsing_models/application_model_v2.py

```diff
@@ -932,14 +983,25 @@ def __init__(
         self._parsed_app_loader = ParsedLogSparkApplicationLoader(None)
         self._unparsed_app_loader = UnparsedLogSparkApplicationLoader(None)
 
-    async def load_raw_datas(self, keys) -> list[DataType]:
+    def _construct_from_parsed_representation(self, key: str, data: tuple[bool, dict]):
+        return self._parsed_app_loader.construct_spark_application(key, data)
+
+    def _construct_from_unparsed_representation(self, key: str, data: tuple[bool, ApplicationModel]):
+        return self._unparsed_app_loader.construct_spark_application(key, data)
+
+    async def _load_raw_datas(self, keys: list[str]) -> list[tuple[bool, dict | ApplicationModel]]:
```

Comment:
This inner implementation, as well as `_construct_base_spark_application`, allow subclasses to take as input more complex objects, but still use the "base" model loading logic given an eventlog location.